### PR TITLE
Send release channel and version in self repair endpoint

### PR DESCRIPTION
**Pull request by mythmon**
@Osmose r?

### CODE REVIEW

File: normandy/selfrepair/static/js/self_repair_runner.js [hunk 1]
```diff
@@ -45,6 +45,36 @@ function loadAction(recipe) {
...
+    let data = {
+        locale: document.documentElement.dataset.locale,
+        user_id: get_user_id(),
+        release_channel: null,
+        version: null,
+    };
+
+    return get_uitour_appinfo()
+    .then(uitour_data => {
+        data['release_channel'] = uitour.defaultUpdateChannel;
```
Comment (Osmose) [inline normandy/selfrepair/static/js/self_repair_runner.js]: Do we prefer `data.release_channel` over `data['release_channel']`?

Comment (mythmon) [inline normandy/selfrepair/static/js/self_repair_runner.js]: Probably. We should really hook up linters to these repos.

File: normandy/selfrepair/static/js/self_repair_runner.js [hunk 2]
```diff
@@ -63,17 +93,12 @@ function get_user_id() {
...
-            user_id: get_user_id(),
-        },
-        headers: {Accept: 'application/json'}
-    }).then(request => {
-        return JSON.parse(request.responseText).recipes;
-    });
+    let {recipeUrl} = document.documentElement.dataset;
+    let headers = {Accept: 'application/json'};
+
+    get_fetch_recipe_payload()
```
Comment (Osmose) [inline normandy/selfrepair/static/js/self_repair_runner.js]: You aren't returning the Promise chain.

File: normandy/selfrepair/static/js/self_repair_runner.js [hunk 3]
```diff
@@ -45,6 +45,36 @@ function loadAction(recipe) {
 }
 
 /**
+ * @promise {Object} The data to send to fetch_bundle to identify this client.
+ */
+function get_fetch_recipe_payload() {
```
Comment (Osmose) [inline normandy/selfrepair/static/js/self_repair_runner.js]: Do we prefer `camelCase` or `snake_case`?

We should get eslint running on this repo eventually.

File: normandy/selfrepair/static/js/self_repair_runner.js (hunk 1 above)
Comment (Osmose) [inline normandy/selfrepair/static/js/self_repair_runner.js]: Also the argument is named `uitour_data` but you're accessing `uitour` here.

**Comment by Osmose** [comment]
It'd be nice to have a test asserting that the parameters are required, but I don't know a non-annoying way to do that, since you'd have to enumerate the invalid cases.

File: normandy/selfrepair/static/js/self_repair_runner.js (hunk 3 above)
Comment (mythmon) [inline normandy/selfrepair/static/js/self_repair_runner.js]: Doh. `camelCase`, most likely. I clearly didn't have all my screws in when I wrote this code.

**Comment by mythmon** [comment]
^ updated

File: normandy/selfrepair/static/js/self_repair_runner.js [hunk 4]
```diff
@@ -45,16 +45,46 @@ function loadAction(recipe) {
...
+        data.version = uitourData.version;
+        return data;
+    });
+}
+
+/**
+ * @promise {Object} The appinfo from UITour
+ */
+function getUitourAppinfo() {
+    return new Promise((resolve, reject) => {
```
Comment (Osmose) [inline normandy/selfrepair/static/js/self_repair_runner.js]: Reject is unused here.

**Comment by Osmose** [comment]
r+

**Pull request merged**